drivers/mssqql.go

```diff
@@ -435,10 +435,10 @@ func (db *MSSQL) ExecuteQuery(query string) ([][]string, error) {
 	}
 
 	if err := rows.Err(); err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
-	return results, nil
+	return results, len(results) - 1, nil
```

Comment:
I don't get the -1 (here and everywhere)

If you want to return the number of records, then it's len(results) only

If you do a -1, then it's an offset.

I'm unsure what are trying to do/get

Maybe I'm simply lost and my comment doesn't make sense

Maybe what you want to do (so something around the -1) has to be done in the code calling these methods and not here

Reply:
Did you read the body of the function? The code is actually really easy to read, so if you go up about 30 lines it should all explain itself.

In short, `results` contains all the rows for the result table so this includes a row with the column names.

Reply:
Indeed, so it's the header. Got it

Reply:
The fact the code is clear don't make obvious why at the end you decrement the number of results.

Maybe you could :
- add a comment
- define a variable where it's about the header, so here do the -1, then return this variable at the end

Apparently, I fail to convince you but I keep trying

Reply:
I think we can do something better here to improve readability.... Will push an update in a minute.